src/main/java/org/takes/facets/auth/PsBasic.java

```diff
@@ -0,0 +1,192 @@
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.facets.auth;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.logging.Level;
+import javax.xml.bind.DatatypeConverter;
+import lombok.EqualsAndHashCode;
+import org.takes.Request;
+import org.takes.Response;
+import org.takes.facets.flash.RsFlash;
+import org.takes.facets.forward.RsForward;
+import org.takes.misc.Opt;
+import org.takes.rq.RqHeaders;
+import org.takes.rs.RsWithHeaders;
+
+/**
+ * Pass that checks the user according RFC-2617.
+ *
+ * <p>The class is immutable and thread-safe.
+ *
+ * @author Endrigo Antonini (teamed@endrigo.com.br)
+ * @version $Id$
+ * @since 0.20
+ * @checkstyle ClassDataAbstractionCouplingCheck (250 lines)
+ */
+@EqualsAndHashCode(of = { "entry", "realm" })
+public final class PsBasic implements Pass {
+
+    /**
+     * Authorization response HTTP head.
+     */
+    private static final String AUTH_HEAD = "Basic";
+
+    /**
+     * Entry to validate user information.
+     */
+    private final transient PsBasic.Entry entry;
+
+    /**
+     * Realm.
+     */
+    private final transient String realm;
+
+    /**
+     * Ctor.
+     * @param rlm Realm
+     * @param basic Entry
+     */
+    public PsBasic(final String rlm, final PsBasic.Entry basic) {
+        this.realm = rlm;
+        this.entry = basic;
+    }
+
+    @Override
+    public Opt<Identity> enter(final Request request) throws IOException {
+        final String decoded = new String(
+            DatatypeConverter.parseBase64Binary(
+                new RqHeaders.Smart(
+                    new RqHeaders.Base(request)
+                ).single("authorization").split(AUTH_HEAD)[1]
+            )
+        ).trim();
+        final String user = decoded.split(":")[0];
+        final Opt<Identity> identity = this.entry.enter(
+            user,
+            decoded.substring(user.length() + 1)
+        );
+        if (!identity.has()) {
+            throw new RsForward(
+                new RsWithHeaders(
+                    new RsFlash("access denied", Level.WARNING),
+                    String.format(
+                        "WWW-Authenticate: Basic ream=\"%s\"",
+                        this.realm
+                    )
+                ),
+                HttpURLConnection.HTTP_UNAUTHORIZED,
+                "/login/start"
```

Comment:
@antonini where did you get this? :) we should use `new RqHref.Base(request).href()` here